diff --git a/node_modules/react-dom-19/cjs/react-dom-client.development.js b/node_modules/react-dom-19/cjs/react-dom-client.development.js
index 83ea1c6..7be6928 100644
--- a/node_modules/react-dom-19/cjs/react-dom-client.development.js
+++ b/node_modules/react-dom-19/cjs/react-dom-client.development.js
@@ -14459,7 +14459,7 @@
                 (lanes & 62914560) === lanes &&
                 ((exitStatus =
                   globalMostRecentFallbackTime +
-                  FALLBACK_THROTTLE_MS -
+                  (globalThis.REACT_FALLBACK_THROTTLE_MS || FALLBACK_THROTTLE_MS) -
                   now$1()),
                 10 < exitStatus)
               ) {
@@ -15596,7 +15596,7 @@
         (workInProgressRootExitStatus === RootSuspended &&
           (workInProgressRootRenderLanes & 62914560) ===
             workInProgressRootRenderLanes &&
-          now$1() - globalMostRecentFallbackTime < FALLBACK_THROTTLE_MS)
+          now$1() - globalMostRecentFallbackTime < (globalThis.REACT_FALLBACK_THROTTLE_MS || FALLBACK_THROTTLE_MS))
           ? (executionContext & RenderContext) === NoContext &&
             prepareFreshStack(root, 0)
           : (workInProgressRootPingedLanes |= pingedLanes),
diff --git a/node_modules/react-dom-19/cjs/react-dom-client.production.js b/node_modules/react-dom-19/cjs/react-dom-client.production.js
index af7283a..78ef4e5 100644
--- a/node_modules/react-dom-19/cjs/react-dom-client.production.js
+++ b/node_modules/react-dom-19/cjs/react-dom-client.production.js
@@ -10391,7 +10391,7 @@ function performWorkOnRoot(root$jscomp$0, lanes, forceSync) {
         shouldTimeSlice.finishedLanes = lanes;
         if (
           (lanes & 62914560) === lanes &&
-          ((renderWasConcurrent = globalMostRecentFallbackTime + 300 - now()),
+          ((renderWasConcurrent = globalMostRecentFallbackTime + (globalThis.REACT_FALLBACK_THROTTLE_MS || 300) - now()),
           10 < renderWasConcurrent)
         ) {
           markRootSuspended(
@@ -11260,7 +11260,7 @@ function pingSuspendedRoot(root, wakeable, pingedLanes) {
     (3 === workInProgressRootExitStatus &&
       (workInProgressRootRenderLanes & 62914560) ===
         workInProgressRootRenderLanes &&
-      300 > now() - globalMostRecentFallbackTime)
+      (globalThis.REACT_FALLBACK_THROTTLE_MS || 300) > now() - globalMostRecentFallbackTime)
       ? 0 === (executionContext & 2) && prepareFreshStack(root, 0)
       : (workInProgressRootPingedLanes |= pingedLanes),
     workInProgressSuspendedRetryLanes === workInProgressRootRenderLanes &&
diff --git a/node_modules/react-dom-19/cjs/react-dom-profiling.development.js b/node_modules/react-dom-19/cjs/react-dom-profiling.development.js
index 7847cbd..78c1fa2 100644
--- a/node_modules/react-dom-19/cjs/react-dom-profiling.development.js
+++ b/node_modules/react-dom-19/cjs/react-dom-profiling.development.js
@@ -14467,7 +14467,7 @@
                 (lanes & 62914560) === lanes &&
                 ((exitStatus =
                   globalMostRecentFallbackTime +
-                  FALLBACK_THROTTLE_MS -
+                  (globalThis.REACT_FALLBACK_THROTTLE_MS || FALLBACK_THROTTLE_MS) -
                   now$1()),
                 10 < exitStatus)
               ) {
@@ -15608,7 +15608,7 @@
         (workInProgressRootExitStatus === RootSuspended &&
           (workInProgressRootRenderLanes & 62914560) ===
             workInProgressRootRenderLanes &&
-          now$1() - globalMostRecentFallbackTime < FALLBACK_THROTTLE_MS)
+          now$1() - globalMostRecentFallbackTime < (globalThis.REACT_FALLBACK_THROTTLE_MS || FALLBACK_THROTTLE_MS))
           ? (executionContext & RenderContext) === NoContext &&
             prepareFreshStack(root, 0)
           : (workInProgressRootPingedLanes |= pingedLanes),
diff --git a/node_modules/react-dom-19/cjs/react-dom-profiling.profiling.js b/node_modules/react-dom-19/cjs/react-dom-profiling.profiling.js
index 98a9f0b..8fae178 100644
--- a/node_modules/react-dom-19/cjs/react-dom-profiling.profiling.js
+++ b/node_modules/react-dom-19/cjs/react-dom-profiling.profiling.js
@@ -10913,7 +10913,7 @@ function performWorkOnRoot(root$jscomp$0, lanes, forceSync) {
         shouldTimeSlice.finishedLanes = lanes;
         if (
           (lanes & 62914560) === lanes &&
-          ((renderWasConcurrent = globalMostRecentFallbackTime + 300 - now$1()),
+          ((renderWasConcurrent = globalMostRecentFallbackTime + (globalThis.REACT_FALLBACK_THROTTLE_MS || 300) - now$1()),
           10 < renderWasConcurrent)
         ) {
           markRootSuspended(
@@ -11881,7 +11881,7 @@ function pingSuspendedRoot(root, wakeable, pingedLanes) {
     (3 === workInProgressRootExitStatus &&
       (workInProgressRootRenderLanes & 62914560) ===
         workInProgressRootRenderLanes &&
-      300 > now$1() - globalMostRecentFallbackTime)
+        (globalThis.REACT_FALLBACK_THROTTLE_MS || 300) > now$1() - globalMostRecentFallbackTime)
       ? 0 === (executionContext & 2) && prepareFreshStack(root, 0)
       : (workInProgressRootPingedLanes |= pingedLanes),
     workInProgressSuspendedRetryLanes === workInProgressRootRenderLanes &&
